File: app/src/main/java/org/eyeseetea/malariacare/data/constant/ServerStaticDataSource.java
package org.eyeseetea.malariacare.data.constant;

import android.content.Context;

import org.eyeseetea.malariacare.R;
import org.eyeseetea.malariacare.data.ReadableServerDataSource;
import org.eyeseetea.malariacare.domain.entity.Server;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

public class ServerStaticDataSource implements ReadableServerDataSource {

    private Context context;

    public ServerStaticDataSource(Context context){
        this.context = context;
    }

    @Override
    public List<Server> getAll() {
        String[] serverUrls = context.getResources().getStringArray(R.array.server_list);
        List<Server> servers = new ArrayList<>();

        for (String url:serverUrls) {
            Server server = new Server(url);
            servers.add(server);
        }

        return servers;
    }

    @NotNull
    @Override
    public Server get() {
        return getAll().get(0);
    }
}

File: app/src/main/java/org/eyeseetea/malariacare/data/constant/ServerStaticDataSource.kt
package org.eyeseetea.malariacare.data.constant

import android.content.Context
import android.util.Log
import org.eyeseetea.malariacare.R
import org.eyeseetea.malariacare.data.ReadableServerDataSource
import org.eyeseetea.malariacare.data.ServerDataSourceFailure
import org.eyeseetea.malariacare.domain.common.Either
import org.eyeseetea.malariacare.domain.entity.Server

class ServerStaticDataSource(private val context: Context) :
    ReadableServerDataSource {
    override fun getAll(): List<Server> {
        Log.d(this.javaClass.simpleName, "Retrieving Servers from static array string")
        val serverUrls = context.resources.getStringArray(R.array.server_list)

        return serverUrls.map { url -> Server(url) }
    }

    override fun get(): Either<ServerDataSourceFailure, Server> { // not implemented
        return Either.Left(ServerDataSourceFailure.ServerNotFoundFailure)
    }
}
